### PR TITLE
Fix dune build of @libs

### PR DESCRIPTION
**Pull request by trefis**
More hacks that should be cleaned after 4.12 is released.

### CODE REVIEW

File: stdlib/dune [hunk 1]
```diff
@@ -29,7 +29,11 @@
           (run sed -i s/loc_FUNCTION/loc_POS/ %{input-file})
           (run awk -v dune_wrapped=true
                  -f %{dep:expand_module_aliases.awk} %{input-file})))
-      stdlib))))
+      stdlib)
+     (; FIXME: remove after 4.12 (this erases injectivity annotations)
+      (action (run sed "s/\\!\\([+]*'\\)/\\1/g" %{input-file}))
```
Comment (gasche) [inline stdlib/dune]: maybe `[+-]`?

**Review by gasche — APPROVED** [review]
Looks good to me.

File: stdlib/dune (hunk 1 above)
Comment (trefis) [reply stdlib/dune]: These aren't present yet, just like `+!'a` isn't used (though it apparently could be, if I believe the manual), and this regex is already ugly enough.

This temporary hack isn't robust, thankfully it truly is only temporary.

**Comment by dra27** [comment]
NB https://github.com/ocaml/ocaml/pull/9843 which has a slightly more comprehensive ocamltest change. I'd dropped the ball on that one, as I got lost in a rabbit hole trying to get `dune build @world` to work - I expect the two PRs combined do that?!

**Comment by trefis** [comment]
> trying to get dune build @world to work - I expect the two PRs combined do that?!

No. :)
Only `@libs` builds, I haven't looked into why linking fails.

> has a slightly more comprehensive ocamltest change

Indeed, the "select" stanza is slightly nicer (although a bit pointless today, since with dune we always build unix).
I merged that PR, will rebase mine and merge.

File: stdlib/dune (hunk 1 above)
Comment (trefis) [reply stdlib/dune]: Since I had to rebase anyway, I changed the regexp as you suggested.

**Pull request merged**